### PR TITLE
Add option to limit project creation

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: Add option to limit project creation

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

```release-note
Add option to limit project creation
```

### CODE REVIEW

File: pkg/crd/kubermatic/v1/settings.go [hunk 1]
```diff
@@ -52,6 +52,7 @@ type SettingSpec struct {
 	DisplayTermsOfService bool           `json:"displayTermsOfService"`
 	EnableDashboard       bool           `json:"enableDashboard"`
 	EnableOIDCKubeconfig  bool           `json:"enableOIDCKubeconfig"`
+	LimitUserProjects     int64          `json:"limitUserProjects"`
```
Comment (maciaszczykm) [inline pkg/crd/kubermatic/v1/settings.go]: LimitUserProject looks like boolean switch. Can you rename to UserProjectsLimit for example?

**Comment by maciaszczykm** [comment]
Thanks.

/lgtm

**Comment by zreigz** [comment]
@kdomanski PTAL

**Pull request merged**